samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -718,15 +719,20 @@ def _get_python_command_name() -> str:
     str
         The name of the python command installed
     """
-    command_names_to_try = ["python", "python3"]
+    command_names_to_try = ["python3", "py3", "python", "py"]
     for command_name in command_names_to_try:
         try:
-            run([command_name, "--version"], check=True, capture_output=True)
+            run_result = run([command_name, "--version"], check=True, capture_output=True, text=True)
         except CalledProcessError:
             pass
         else:
+            # check for python 3, 3.7 or above
+            # regex: search for 'Python', whitespace, '3.', digits 7-9 or 2+ digits, any digit or '.' 0+ times
+            match = re.search(r"Python\s*3.([7-9]|\d{2,})[\d.]*", run_result.stdout)
+            if not match:
+                continue
             return command_name
-    raise PrepareHookException("Python not found. Please ensure that python is installed.")
+    raise PrepareHookException("Python not found. Please ensure that python 3.7 or above is installed.")
```

Comment:
We need to have doc updates to ensure this is a pre-req to use.

Reply:
It's the same version supported by the sam cli (setup.py)